File: CS/CustomDataExplorer/MyMenuCommandHandler.cs
using DevExpress.Snap;

namespace CustomDataExplorer {
    public class MyMenuCommandHandler : MenuCommandHandler {
        public MyMenuCommandHandler(SnapControl snap) : base(snap) {
        }
        public override void UpdateCommandStatus() {
            base.UpdateCommandStatus();
            foreach (DevExpress.XtraReports.Design.Commands.CommandSetItem item in commands) {
                if (item.CommandID == DataExplorerCommands.AddDataSource) {
                    item.Supported = false;
                }
            }
        }
    }
}

File: VB/CustomDataExplorer/MyMenuCommandHandler.vb
Imports DevExpress.Snap

Namespace CustomDataExplorer
    Public Class MyMenuCommandHandler
        Inherits MenuCommandHandler

        Public Sub New(ByVal snap As SnapControl)
            MyBase.New(snap)
        End Sub
        Public Overrides Sub UpdateCommandStatus()
            MyBase.UpdateCommandStatus()
            For Each item As DevExpress.XtraReports.Design.Commands.CommandSetItem In commands
                If item.CommandID Is DataExplorerCommands.AddDataSource Then
                    item.Supported = False
                End If
            Next item
        End Sub
    End Class
End Namespace